e2e/tests/core/02-client/client_test.go

```diff
@@ -52,6 +52,14 @@ type ClientTestSuite struct {
 	testsuite.E2ETestSuite
 }
 
+func (s *ClientTestSuite) SetupTest() {
+	ctx := context.TODO()
+	chainA, chainB := s.GetChains()
```

Comment:
one thing I notice here, is that `s.GetChains()` allows for an optional function to be provided to modify the chain configurations, with these current `SetupTest` functions, the default options are used in all cases. I think this isn't the end of the world as custom test suites can provide their own `SetupTest` functions when needed, but worth mentioning I think.

Reply:
Hmm, I think in special cases where we need to modify chainOption, we will have to move it to a separate suite. perhaps that's the trade-off with the current solution.